linalg/densemat.hpp

```diff
@@ -474,10 +494,19 @@ class LUFactors
 
    LUFactors(double *data_, int *ipiv_) : data(data_), ipiv(ipiv_) { }
 
-   /** Factorize the current data of size (m x m) overwriting it with the LU
-       factors. The factorization is such that L.U = P.A, where A is the
-       original matrix and P is a permutation matrix represented by ipiv. */
-   void Factor(int m);
+   /**
+    * @brief Compute the LU factorization of the current matrix
+    *
+    * Factorize the current matrix of size (m x m) overwriting it with the
+    * LU factors. The factorization is such that L.U = P.A, where A is the
+    * original matrix and P is a permutation matrix represented by ipiv.
+    *
+    * @param [in] m size of the square matrix
+    * @param [in] TOL optional fuzzy comparison tolerance. Defaults to 1e-9.
+    *
+    * @return status set to true if successful, otherwise, false.
+    */
+   bool Factor(int m, double TOL = 1.e-9);
```

Comment:
I am ready to merge this, but I will like to make one final change: can we set the default `TOL` value in `Factor` to `0.0` so we don't change the behavior in existing user code? (It is OK to have a different value in `LinearSolve`)

Reply:
Certainly, I'll make the change.

Reply:
Done.